pokeapp/src/components/FilterBar.js

```diff
@@ -0,0 +1,34 @@
+const FilterBar = ({ allPokemon, setAllPokemon }) => {
+
+  const sortAtoZ = () => {
+    setAllPokemon(
+      allPokemon.sort((pokemon1, pokemon2) => {
+        if (pokemon1.name < pokemon2.name) return -1;
+        if (pokemon1.name > pokemon2.name) return 1;
+        return 0;
+      })
+    );
+  };
+  const sortZtoA = () => {
+    setAllPokemon(
+      allPokemon.sort((pokemon1, pokemon2) => {
+        if (pokemon1.name < pokemon2.name) return 1;
+        if (pokemon1.name > pokemon2.name) return -1;
+        return 0;
```

Comment:
I'd advise against overwriting the original copy.  The reason for this is that if you want to apply filters or sorting to the data, it's easier to remove them if you have a separate copy of the mutated data.  I recommend keeping the original copy static, then using that original copy to _derive_ the mutated version based on sort and filter.

Reply:
In fact, if you apply filters then overwrite the original, you'll run into the problem that I fixed on a client project last year.  The team was applying filters to the original copy of the data, which made it impossible to remove the filters without performing a whole API call to restore the original.